aeron-samples/src/main/java/io/aeron/samples/archive/EmbeddedRecordingThroughput.java

```diff
@@ -99,12 +101,12 @@ public EmbeddedRecordingThroughput()
 
     public void close()
     {
-        CloseHelper.close(aeronArchive);
-        CloseHelper.close(aeron);
-        CloseHelper.close(archivingMediaDriver);
-
-        archivingMediaDriver.archive().context().deleteArchiveDirectory();
-        archivingMediaDriver.mediaDriver().context().deleteAeronDirectory();
+        CloseHelper.closeAll(
+            aeronArchive,
+            aeron,
+            archivingMediaDriver,
+            () -> archivingMediaDriver.archive().context().deleteArchiveDirectory(),
```

Comment:
We should add a delete specific function that takes an error handler. This would avoid quite a few lambdas.

Reply:
@mjpt777 Having a delete with an `ErrorHandler` won't help in this case since we are deleting the aeron directories which contain the error log to which `ErrorHandler` will normally write. So the only thing we can do if don't want to use lambdas when deleting directories is to leverage the fact that the `org.agrona.IoUtil#delete` method accepts a boolean which allows ignoring errors.
So the `close()` might become:
```java
        CloseHelper.closeAll(
            aeronArchive,
            aeron,
            archivingMediaDriver);
       IOUtil.delete(new File(archivingMediaDriver.archive().context().deleteArchiveDirectory()), true);
       IOUtil.delete(new File(archivingMediaDriver.mediaDriver().context().deleteAeronDirectory()), true);
``` 
However the deletes will not be performed if the `closeAll` throws. Which means we will also need to use `quiteCloseAll` instead.

Provided all of the above and the fact that this is all in samples I'm keen on keeping the current implementation.

Reply:
We need to make sure the delete is attempted and any errors reported. Lambdas may be required here.

Reply:
@mjpt777 OK, so I'll keep current logic.